src/NServiceBus.Testing/Sagas/TestableSaga.cs

```diff
@@ -216,31 +221,32 @@ async Task<MessageProcessingResult> InnerProcess(QueuedSagaMessage message, stri
 
             EnqueueMessagesAndTimeouts(context, saga.Entity.Id);
 
-            return new MessageProcessingResult(saga, message, context);
+            return new HandleResult(saga, message, context);
         }
 
         /// <summary>
-        /// Advance the <see cref="CurrentTime"/> for the test, dispatching any stored timeouts along the way.
+        /// Advance the <see cref="CurrentTime"/> for the saga and handle any timeouts due during <paramref name="timeToAdvance"/>.
         /// </summary>
         /// <param name="timeToAdvance">Amount of time to advance the <see cref="CurrentTime"/>.</param>
         /// <param name="cancellationToken">A cancellation token.</param>
         /// <returns>
-        /// Returns an array of <see cref="MessageProcessingResult"/>, one for each timeout that was dispatched and processed
-        /// during the time that was advanced. If no timeouts were fired, the array will be empty.
+        /// Returns an array of <see cref="HandleResult"/>,
+        /// one for each timeout handled during <paramref name="timeToAdvance"/>.
+        /// If no timeouts were handled, the array is empty.
         /// </returns>
-        public async Task<MessageProcessingResult[]> AdvanceTime(TimeSpan timeToAdvance, CancellationToken cancellationToken = default)
+        public async Task<HandleResult[]> AdvanceTime(TimeSpan timeToAdvance, CancellationToken cancellationToken = default)
```

Comment:
It _feels_ like this should accept a `TestableMessageHandlerContext`, similar to the `Handle*` methods, but I guess that might introduce some complexity?